pytorch_lightning/callbacks/batch_size_finder.py

```diff
@@ -0,0 +1,319 @@
+                    garbage_collection_cuda()
+                    high = new_size
+                    midval = (high + low) // 2
+                    new_size, _ = self._adjust_batch_size(trainer, value=midval, desc="failed")
+                    if high - low <= 1:
+                        break
+                else:
+                    raise  # some other error not memory related
+
+        return new_size
+
+    def _try_loop_run(self, trainer):
+        if trainer.state.fn == TrainerFn.FITTING:
+            trainer.fit_loop.global_step = self._dumped_params["global_step"]
+            trainer.fit_loop.current_epoch = self._dumped_params["current_epoch"]
+            trainer.fit_loop.run()
+        elif trainer.state.fn == TrainerFn.VALIDATING:
+            trainer.validate_loop.run()
+        elif trainer.state.fn == TrainerFn.TESTING:
+            trainer.test_loop.run()
+        elif trainer.state.fn == TrainerFn.PREDICTING:
+            trainer.predict_loop.run()
+
+    @staticmethod
+    def _reset_dataloaders(trainer, pl_module):
+        if trainer.state.fn == TrainerFn.FITTING:
+            trainer.reset_train_dataloader(pl_module)
+            trainer.reset_val_dataloader(pl_module)
+        elif trainer.state.fn == TrainerFn.VALIDATING:
+            trainer.reset_val_dataloader(pl_module)
+        elif trainer.state.fn == TrainerFn.TESTING:
+            trainer.reset_test_dataloader(pl_module)
+        elif trainer.state.fn == TrainerFn.PREDICTING:
+            trainer.reset_predict_dataloader(pl_module)
+
+    def _dump_params(self, trainer):
+        self._dumped_params = {
+            "current_epoch": trainer.current_epoch,
+            "global_step": trainer.global_step,
+            "max_steps": trainer.max_steps,
+            "logger": trainer.logger,
+            "callbacks": trainer.callbacks,
+            "limit_train_batches": trainer.limit_train_batches,
+            "limit_val_batches": trainer.limit_val_batches,
+            "limit_test_batches": trainer.limit_test_batches,
+            "limit_predict_batches": trainer.limit_predict_batches,
+        }
+
+    def _reset_params(self, trainer):
+        trainer.logger = DummyLogger() if trainer.logger is not None else None
+        trainer.callbacks = []
+        if trainer.state.fn == TrainerFn.FITTING:
+            trainer.limit_val_batches = self.steps_per_trial
+            trainer.fit_loop.max_steps = self.steps_per_trial
+        elif trainer.state.fn == TrainerFn.VALIDATING:
+            trainer.limit_val_batches = self.steps_per_trial
+        elif trainer.state.fn == TrainerFn.TESTING:
+            trainer.limit_test_batches = self.steps_per_trial
+        elif trainer.state.fn == TrainerFn.PREDICTING:
+            trainer.limit_predict_batches = self.steps_per_trial
+
+    def _restore_params(self, trainer):
+        trainer.fit_loop.current_epoch = self._dumped_params["current_epoch"]
```

Comment:
I am not sure if this would clean properly the progress objects.

Reply:
I think saving and restoring loop state dict should work fine. wdyt?